k8s/src/main/scala/io/buoyant/k8s/EndpointsNamer.scala

```diff
@@ -28,25 +27,23 @@ class EndpointsNamer(
     case id@Path.Utf8(nsName, portName, serviceName) =>
       val residual = path.drop(PrefixLen)
       log.debug("k8s lookup: %s %s", id.show, path.show)
-      Ns.get(nsName).services.flatMap { services =>
-        log.debug("k8s ns %s initial state: %s", nsName, services.keys.mkString(", "))
-        services.get(serviceName) match {
+      Ns.get(nsName).map { nsCache =>
+        log.debug("k8s ns %s initial state: %s", nsName, nsCache.services.keys.mkString(", "))
+        nsCache.services.get(serviceName) match {
           case None =>
             log.debug("k8s ns %s service %s missing", nsName, serviceName)
-            Activity.value(NameTree.Neg)
+            NameTree.Neg
 
           case Some(services) =>
             log.debug("k8s ns %s service %s found", nsName, serviceName)
-            services.ports.map { ports =>
-              ports.get(portName) match {
-                case None =>
-                  log.debug("k8s ns %s service %s port %s missing", nsName, serviceName, portName)
-                  NameTree.Neg
+            services.ports.get(portName) match {
+              case None =>
+                log.debug("k8s ns %s service %s port %s missing", nsName, serviceName, portName)
+                NameTree.Neg
 
-                case Some(port) =>
-                  log.debug("k8s ns %s service %s port %s found + %s", nsName, serviceName, portName, residual.show)
-                  NameTree.Leaf(Name.Bound(port.addr, idPrefix ++ id, residual))
-              }
+              case Some(port) =>
+                log.debug("k8s ns %s service %s port %s found + %s", nsName, serviceName, portName, residual.show)
+                NameTree.Leaf(Name.Bound(Var(port.addr), idPrefix ++ id, residual))
```

Comment:
I think what this exercise has led me to is that I think I don't understand why finagle has both an Activity for the NameTree and a Var for the Addr.  Just watching the addr Var directly seems problematic because a change to the NameTree could invalidate that Var.  Therefore, it seems like you would always want to watch the NameTree flatMapped to the addr... in which case, why have both?